src/support/z_file_locate.erl

```diff
@@ -234,7 +235,7 @@ generate_preview(true, _Mime, Path, OriginalFile, Filters, Medium, Context) ->
             PreviewFilePath = filename:join(PreviewDir, Path),
             case z_media_preview:convert(z_convert:to_list(Filename), OriginalFile, z_convert:to_list(PreviewFilePath), Filters, Context) of
                 ok ->
-                    FileStorePath = filename:join([filename:basename(PreviewDir), Path]),
+                    FileStorePath = z_convert:to_binary(filename:join([filename:basename(PreviewDir), Path])),
                     z_notifier:first(#filestore{action=upload, path=FileStorePath}, Context),
                     case proplists:get_value(id, Medium) of
```

Comment:
The expression on line 240 and column 21 is nested beyond the maximum level of 3.